src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -245,17 +246,25 @@ def _generate_external_and_total_bills(self, area, results, flattened):
             # External trades are the trades of the parent area
             external = flattened[area.name].copy()
             # Should not include market fee to the external trades
-            market_fee = external.pop("market_fee", None)
+            market_fee = external.pop("market_fee", 0.)
             # Should switch spent/earned and bought/sold, to match the perspective of the UI
-            spent = external.pop("spent")
+            spent = external.pop("spent") + market_fee
             earned = external.pop("earned")
             bought = external.pop("bought")
             sold = external.pop("sold")
             external.update(**{"spent": earned, "earned": spent, "bought": sold,
-                               "sold": bought, "market_fee": market_fee})
+                               "sold": bought, "market_fee": 0.})
             external["total_energy"] = external["bought"] - external["sold"]
             external["total_cost"] = external["spent"] - external["earned"]
             results[area.name].update({"External Trades": external})
+            results[area.name]["Accumulated Trades"]["bought"] += external["bought"]
```

Comment:
To summarize our discussion on slack, it is better if the Accumulated Trades are not recalculated here again. All that is needed is to take into account the External Trades row when calculating the Accumulated Trades row. In order to achieve that the calculation order of these 2 rows needs to be swapped, essentially to calculate first the External Trades row and after that the Accumulated Trades row (it involves just moving the external trade calculation code above the accumulated trades).